apps/meteor/app/lib/server/functions/saveUser.ts

```diff
@@ -39,43 +46,41 @@ Meteor.startup(() => {
 	});
 });
 
-async function _sendUserEmail(subject, html, userData) {
+async function _sendUserEmail(subject: string, html: string, userData: RequiredField<ISaveUserDataParams, 'email'>) {
 	const email = {
 		to: userData.email,
-		from: settings.get('From_Email'),
+		from: settings.get<string>('From_Email'),
 		subject,
 		html,
 		data: {
 			email: userData.email,
 			password: userData.password,
+			...(userData.name && { name: userData.name }),
 		},
 	};
 
-	if (typeof userData.name !== 'undefined') {
-		email.data.name = userData.name;
-	}
-
 	try {
 		await Mailer.send(email);
-	} catch (error) {
+	} catch (error: any) {
 		throw new Meteor.Error('error-email-send-failed', `Error trying to send email: ${error.message}`, {
 			function: 'RocketChat.saveUser',
 			message: error.message,
 		});
 	}
 }
 
-async function validateUserData(userId, userData) {
+async function validateUserData(userId: IUser['_id'], userData: ISaveUserDataParams) {
 	const existingRoles = _.pluck(await getRoles(), '_id');
+	const isUpdateUserParams = '_id' in userData && userData._id;
```

Comment:
```suggestion
	const isUpdateUserParams = userData._id;
```
would do the same thing, since not narrowing the type here shouldn't be needed.